tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/QuartzTests.cs

```diff
@@ -0,0 +1,111 @@
+
+    [SkippableTheory]
+    [Trait("Category", "EndToEnd")]
+    [MemberData(nameof(PackageVersions.Quartz), MemberType = typeof(PackageVersions))]
+    public async Task SubmitsTraces(string packageVersion)
+    {
+        SetEnvironmentVariable("DD_TRACE_OTEL_ENABLED", "true");
+
+        using (var telemetry = this.ConfigureTelemetry())
+        using (var agent = EnvironmentHelper.GetMockAgent())
+        using (await RunSampleAndWaitForExit(agent, packageVersion: packageVersion))
+        {
+            var filename = nameof(QuartzTests) + GetSuffix(packageVersion);
+            Log.Information(filename);
+            const int expectedSpanCount = 2;
+            var spans = await agent.WaitForSpansAsync(expectedSpanCount);
+
+            using var s = new AssertionScope();
+            spans.Count.Should().Be(expectedSpanCount);
+
+            var myServiceNameSpans = spans.Where(s => s.Service == "Samples.Quartz");
+            ValidateIntegrationSpans(myServiceNameSpans, metadataSchemaVersion: "v0", expectedServiceName: "Samples.Quartz", isExternalSpan: false);
+
+            var settings = VerifyHelper.GetSpanVerifierSettings();
+            var traceStatePRegex = new Regex("p:[0-9a-fA-F]+");
+            var traceIdRegexHigh = new Regex("TraceIdLow: [0-9]+");
+            var traceIdRegexLow = new Regex("TraceIdHigh: [0-9]+");
+            var fireInstanceId = new Regex(@"fire\.instance\.id:\s*\d+");
+            var scrubErrorEvents = new Regex(@"events:\s*\[.*?\]");
+            var scrubOtelVersion = new Regex(@"otel\.library\.version:\s*[\d\.]+");
+            settings.AddRegexScrubber(traceStatePRegex, "p:TsParentId");
+            settings.AddRegexScrubber(traceIdRegexHigh, "TraceIdHigh: LinkIdHigh");
+            settings.AddRegexScrubber(traceIdRegexLow, "TraceIdLow: LinkIdLow");
+            settings.AddRegexScrubber(_timeUnixNanoRegex, @"time_unix_nano"":<DateTimeOffset.Now>");
+            settings.AddRegexScrubber(_versionRegex, "telemetry.sdk.version: sdk-version");
+            settings.AddRegexScrubber(fireInstanceId, "fire.instance.id: <fire.instance.id>");
+            settings.AddRegexScrubber(scrubErrorEvents, "events: <error_stack>");
+            settings.AddRegexScrubber(scrubOtelVersion, "otel.library.version: <otel-library-version>");
+
+            await VerifyHelper.VerifySpans(
+                                        spans,
+                                        settings,
+                                        orderSpans: s => s
+                                                       .OrderBy(x => x.Name)
+                                                       .ThenBy(x => x.Resource))
+                              .UseFileName(filename);
+
+            await telemetry.AssertIntegrationEnabledAsync(IntegrationId.OpenTelemetry);
+        }
+    }
+
+    private static string GetSuffix(string packageVersion)
```

Comment:
nit: totally optional, but "switch expressions" are awesome, and make things much easier to read IMO 😄


```csharp
    private static string GetSuffix(string packageVersion)
    {
        if (string.IsNullOrEmpty(packageVersion))
        {
            return "V4";
        }

        return new Version(packageVersion) switch
        {
            { } v when v >= new Version("4.0.0") => "V4",
    #if NETCOREAPP3_1
            { } v when v >= new Version("3.15.0") => "V315plusNETCOREAPP31",
    #endif
            _ => "V3",
        }
    }
```